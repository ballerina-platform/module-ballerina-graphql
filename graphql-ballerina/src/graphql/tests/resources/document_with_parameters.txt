query getData {
    # Line comment
    name(id: 1, name: sam) # Inline Comment 
    id
    birthdate
}
