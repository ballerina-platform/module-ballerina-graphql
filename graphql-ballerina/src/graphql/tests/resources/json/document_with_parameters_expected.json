{
    "operations": {
        "operation": {
            "name": "getData",
            "type": "query",
            "fields": [
                {
                    "name": "name",
                    "arguments": [
                        {
                            "name": "id",
                            "value": 132,
                            "type": "INLINE",
                            "nameLocation": {
                                "line": 8,
                                "column": 10
                            },
                            "valueLocation": {
                                "line": 8,
                                "column": 14
                            }
                        },
                        {
                            "name": "name",
                            "value": "Prof. Moriarty",
                            "type": "INLINE",
                            "nameLocation": {
                                "line": 8,
                                "column": 19
                            },
                            "valueLocation": {
                                "line": 8,
                                "column": 25
                            }
                        },
                        {
                            "name": "negative",
                            "value": -123,
                            "type": "INLINE",
                            "nameLocation": {
                                "line": 8,
                                "column": 43
                            },
                            "valueLocation": {
                                "line": 8,
                                "column": 53
                            }
                        },
                        {
                            "name": "weight",
                            "value": 75.4,
                            "type": "INLINE",
                            "nameLocation": {
                                "line": 8,
                                "column": 59
                            },
                            "valueLocation": {
                                "line": 8,
                                "column": 67
                            }
                        }
                    ],
                    "selections": [
                        {
                            "name": "first",
                            "location": {
                                "line": 9,
                                "column": 9
                            }
                        }
                    ],
                    "location": {
                        "line": 8,
                        "column": 5
                    }
                },
                {
                    "name": "id",
                    "selections": [
                        {
                            "name": "prefix",
                            "selections": [
                                {
                                    "name": "sample",
                                    "location": {
                                        "line": 14,
                                        "column": 13
                                    }
                                }
                            ],
                            "location": {
                                "line": 13,
                                "column": 9
                            }
                        },
                        {
                            "name": "suffix",
                            "location": {
                                "line": 16,
                                "column": 9
                            }
                        }
                    ],
                    "location": {
                        "line": 12,
                        "column": 5
                    }
                },
                {
                    "name": "birthdate",
                    "arguments": [
                        {
                            "name": "format",
                            "value": "DD/MM/YYYY",
                            "type": "INLINE",
                            "nameLocation": {
                                "line": 18,
                                "column": 16
                            },
                            "valueLocation": {
                                "line": 18,
                                "column": 24
                            }
                        }
                    ],
                    "location": {
                        "line": 18,
                        "column": 5
                    }
                }
            ]
        }
    }
}
